firedrake_mlmc/ensemble_hierarchy.py

```diff
@@ -115,6 +132,54 @@ def __UpdateDomain(self, s):
 
         return Tuple_to_append
 
+    def UpdateEnsembleMember(self, S):
+
```

Comment:
Think about how we would do this is the assimilation context. The particles change/modified and can be rearranged, so consider this case and his we make sure identifier and index stay with the actual object of the state and not the 'particle'.